packages/react/src/analytics/integrations/shared/GoogleConsentMode/GoogleConsentMode.ts

```diff
@@ -0,0 +1,124 @@
+import { type ConsentData } from '@farfetch/blackout-analytics';
+import { type GoogleConsentModeMappings } from './types.js';
+
+/**
+ * GoogleConsentMode handles with Google Consent Mode v2.
+ */
+export class GoogleConsentMode {
+  private dataLayers!: Array<string>; // Stores different data layer names
+  private enabled!: boolean; // enable or disable CMP integration
+  private consentCategoriesMapping?: GoogleConsentModeMappings; // Stores default or customized consent category mappings
+
+  constructor() {
+    this.dataLayers = [];
+    this.enabled = true;
+  }
+
+  /**
+   * Adds an integration for a specific data layer with a given consent.
+   *
+   * @param consent - The consent data.
+   * @param dataLayer - The data layer name.
+   */
+  addIntegration(consent: ConsentData | null, dataLayer: string) {
+    if (this.enabled) {
+      this.dataLayers.push(dataLayer);
+      this.gtag(dataLayer, consent);
+    }
+  }
+
+  /**
+   * Sets custom mappings for consent categories.
+   *
+   * @param consentMapping - Custom mappings to be set.
+   */
+  setConsentCategoriesMapping(consentMapping: GoogleConsentModeMappings) {
+    this.consentCategoriesMapping = consentMapping;
+  }
+
+  /**
+   * Disables Google Consent Mode.
+   */
+  disable() {
+    this.enabled = false;
+  }
+
+  /**
+   * Retrieves consent values for each category based on provided consents.
+   *
+   * @param consents - The consent data.
+   *
+   * @returns Consent values for each category.
+   */
+  private getConsentValues(consents: ConsentData) {
```

Comment:
Use singular here please, since it is the same data type that is being used in setConsent throughout the package.